src/modules/git_status.rs

```diff
@@ -665,6 +676,25 @@ mod tests {
         repo_dir.close()
     }
 
+    #[test]
+    fn shows_no_upstream_branch() -> io::Result<()> {
+        let repo_dir = fixture_repo(FixtureProvider::Git)?;
+
+        new_branch(repo_dir.path())?;
+
+        let actual = ModuleRenderer::new("git_status")
+            .config(toml::toml! {
+                [git_status]
+                no_upstream="🥷"
+            })
+            .path(repo_dir.path())
+            .collect();
+        let expected = format_output("🥷");
+
+        assert_eq!(expected, actual);
```

Comment:
This test is failing (the value of `actual` is `None` rather than the expected string), and I don't know how to fix it.

Is it a bug in the implementation itself, or in the test? I built this test based on the other ones, so I assumed it would be similar, but I never coded Rust before, so it's possible that the test is incorrect. Any help appreciated! :sweat:

![image](https://github.com/starship/starship/assets/478237/6c0c3522-f279-4c88-831a-6b2286a1cfea)

Reply:
I would want the test to pass before reviewing this in more detail.

Reply:
Right, me too, but that's precisely where I'm struck at and asking for help with. Would it help if I described what I tried to accomplish with the code changes in natural language or pseudocode?

Reply:
Seems like that the failure is directly related to the functionality not being available. Have you tried these changes locally and got the expected result?